File: patch.py
import os
import re

dir_path = 'frontend/src'
for root, dirs, files in os.walk(dir_path):
    for file in files:
        if file.endswith('.tsx') or file.endswith('.ts'):
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                content = f.read()
            
            # Use regex to match fetch('/api and fetch(/api
            new_content = re.sub(r\"fetch\(\s*'/api/v1\", r\"fetch((import.meta.env.VITE_API_URL || '') + '/api/v1\", content)
            new_content = re.sub(r\"fetch\(\s*\/api/v1\", r\"fetch(\${import.meta.env.VITE_API_URL || ''}/api/v1\", new_content)

            if new_content != content:
                with open(path, 'w', encoding='utf-8') as f:
                    f.write(new_content)
                print('Updated ' + path)

File: patch.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        file = path.join(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            results = results.concat(walk(file));
        } else {
            if (file.endsWith('.tsx') || file.endsWith('.ts')) {
                results.push(file);
            }
        }
    });
    return results;
}

const files = walk('frontend/src');
files.forEach(file => {
    let content = fs.readFileSync(file, 'utf8');
    let original = content;
    
    content = content.replace(/fetch\(\s*\'\/api\/v1/g, "fetch((import.meta.env.VITE_API_URL || '') + '/api/v1");
    content = content.replace(/fetch\(\s*\`\/api\/v1/g, "fetch(`${import.meta.env.VITE_API_URL || ''}/api/v1");
    
    if (content !== original) {
        fs.writeFileSync(file, content, 'utf8');
        console.log('Updated ' + file);
    }
});
